tools/astarte_import/lib/astarte/import.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Astarte.Import do
+  defmodule State do
+    defstruct [
+      :device_id,
+      :interface,
+      :path,
+      :timestamp,
+      :insert_value_fun,
+      :data
+    ]
+  end
+
+  def parse(xml, fun, cont_fun) do
+    state = %State{
+      insert_value_fun: fun
+    }
+
+    opts = [event_fun: &xml_event/3, continuation_fun: cont_fun, event_state: state]
+
+    {:ok, state, _tail} = :xmerl_sax_parser.stream(xml, opts)
+
+    state.data
+  end
+
+  defp xml_event({:startElement, _uri, _l_name, {_prefix, 'astarte'}, _attributes}, _loc, state) do
+    state
+  end
+
+  defp xml_event({:startElement, _uri, _l_name, {_prefix, 'devices'}, _attributes}, _loc, state) do
+    state
+  end
+
+  defp xml_event({:startElement, _uri, _l_name, {_prefix, 'device'}, attributes}, _loc, state) do
+    {:ok, device_id} = get_attribute(attributes, 'device_id')
+
+    %State{state | device_id: device_id}
+  end
+
+  defp xml_event(
+         {:startElement, _uri, _l_name, {_prefix, 'interfaces'}, _attributes},
+         _loc,
+         state
+       ) do
+    state
+  end
+
+  defp xml_event({:startElement, _uri, _l_name, {_prefix, 'interface'}, attributes}, _loc, state) do
+    with {:ok, name} <- get_attribute(attributes, 'name'),
+         {:ok, major_string} <- get_attribute(attributes, 'version_major'),
+         {major, ""} <- Integer.parse(major_string),
+         {:ok, minor_string} <- get_attribute(attributes, 'version_minor'),
```

Comment:
Use `minor_version` to match how we store things in the `Astarte.Core.Interface` struct